### PR TITLE
fix(curriculum):Updated lessons in Number sorter project to better show how ``inputValues`` works

### LINKED ISSUE #54434: Update lessons in number sorter project to better show how `inputValues` works
### Describe the Issue

In [step 7](https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-basic-algorithmic-thinking-by-building-a-number-sorter/step-7) of the project, campers create this `inputValues` array

```js
 const inputValues = [
    ...document.getElementsByClassName("values-dropdown")
  ].map((dropdown) => Number(dropdown.value));
```
I think it would be more beneficial to show them the original string inputs and then walk them through converting it to numbers. 

Here is my proposed plan:

## updated step 7
This step should just focus on having campers use map to return dropdown.value
 
```
   const inputValues = [
    ...document.getElementsByClassName("values-dropdown")
  ].map((dropdown) => dropdown.value);
```

## new step 8
This new step should have campers log out the `inputValues` so they can see for themselves that the values are strings and not numbers

```
 console.log(inputValues)
```

## new step 9
For this new step, this is when they update the `inputValues` map method to use `Number()`. Have campers check the console again to see that we are no longer dealing with strings but now dealing with numbers

```
  const inputValues = [
    ...document.getElementsByClassName("values-dropdown")
  ].map((dropdown) => Number(dropdown.value));
```

## new step 10
This step should be to just remove the console statement.

```
  console.log(inputValues)

```

Then the project resumes as normal with new step 10 which is just this step [here](https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-basic-algorithmic-thinking-by-building-a-number-sorter/step-8)

If you are new to adding steps, please read through the documentation here 
https://contribute.freecodecamp.org/#/how-to-work-on-practice-projects?id=how-to-work-on-practice-projects

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-basic-algorithmic-thinking-by-building-a-number-sorter/step-7

### Your code

see explanation above

### Expected behavior

see explanation above

### Screenshots

_No response_

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by sanskriti2005**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #54434

### CODE REVIEW

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-algorithmic-thinking-by-building-a-number-sorter/662236690376072dc79d9c27.md [hunk 1]
```diff
@@ -0,0 +1,325 @@
+---
+id: 662236690376072dc79d9c27
+title: Step 10
+challengeType: 0
+dashedName: step-10
+---
+
+# --description--
+
+Remove the `console.log()` calling on the `inputValues`.
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-algorithmic-thinking-by-building-a-number-sorter/662236690376072dc79d9c27.md:10]:
```suggestion
Now that you have confirmed the data type of the `inputValues` elements, remove your `console.log()` call.
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-algorithmic-thinking-by-building-a-number-sorter/6407b940b8983005578d0824.md [hunk 2]
```diff
@@ -7,9 +7,9 @@ dashedName: step-7
 
 # --description--
 
-You need to get the values from your `select` elements. However, these values are strings, and you need them to be numbers.
+You need to get the values from your `select` elements. These values will currenly be strings and you will convert them into numbers.
```
Comment (Ksound22) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-algorithmic-thinking-by-building-a-number-sorter/6407b940b8983005578d0824.md:10]:
```suggestion
You need to get the values from your `select` elements. These values will currently be strings and you will convert them into numbers.
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-algorithmic-thinking-by-building-a-number-sorter/6621c5f02a5c272761db8661.md [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+```
+
+In the next step, you will convert those strings into numbers.
+
+# --hints--
+
+You should call `console.log()`.
+
+```js
+assert.match(code, /console\.log\(.*\);/);
```
Comment (Ksound22) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-algorithmic-thinking-by-building-a-number-sorter/6621c5f02a5c272761db8661.md:27]: The semi-colon here should be optional. If a camper fails to use the semi-colon, the test will not pass, so the next hint and test won't be reached.
 
```suggestion
assert.match(code, /console\.log\(.*\);?/);
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-algorithmic-thinking-by-building-a-number-sorter/6621c5f02a5c272761db8661.md [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+```
+
+You should pass `inputValues` to your `console.log()` call.
+
+```js
+assert.match(code, /console\.log\(\s*inputValues\s*\);/);
```
Comment (Ksound22) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-algorithmic-thinking-by-building-a-number-sorter/6621c5f02a5c272761db8661.md:33]:
```suggestion
assert.match(code, /console\.log\(\s*inputValues\s*\);?/);
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-algorithmic-thinking-by-building-a-number-sorter/6621c5f02a5c272761db8661.md [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+
+Before going further, make sure to observe the data type of the printed result in the console. 
+
+To see the result for the `inputValues` array, log out the `inputValues` array to the console. Then click on the sort button and open up the console. You should see an array of strings like this:
+
+```js
+[ '8', '2', '4', '1', '3' ]
+```
+
+In the next step, you will convert those strings into numbers.
```
Comment (Ksound22) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-algorithmic-thinking-by-building-a-number-sorter/6621c5f02a5c272761db8661.md:20]:
```suggestion
Before going further, make sure you observe the data type of the printed result in the console.

In the next step, you will convert those strings into numbers.
```

**Review by Ksound22 — COMMENTED** [review]
I left some comments. Other than those, I think things look good.

I will approve when you implement the changes.

**Comment by sanskriti2005** [comment]
@Ksound22 Thank you for the review! I will make the changes

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-algorithmic-thinking-by-building-a-number-sorter/6621c5f02a5c272761db8661.md [hunk 6]
```diff
@@ -0,0 +1,342 @@
...
+
+Before going further, make sure to observe the data type of the printed result in the console. 
+
+To see the result for the `inputValues` array, log out the `inputValues` array to the console. Then click on the sort button and open up the console. You should see an array of strings like this:
+
+```js
+[ '8', '2', '4', '1', '3' ]
+```
+
+Before going further, make sure you observe the data type of the printed result in the console.
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-algorithmic-thinking-by-building-a-number-sorter/6621c5f02a5c272761db8661.md:20]:
````suggestion
To see the result for the `inputValues` array, log out the `inputValues` array to the console. Then click on the sort button and open up the console. You should see an array of strings like this:

```js
[ '8', '2', '4', '1', '3' ]
```

Before going further, make sure you observe the data type of the printed result in the console.
````

Let's not repeat ourselves, though.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

**Review by naomi-lgbt — APPROVED** [review]
Thanks for all of your work on this.

**Review by jdwilkin4 — APPROVED** [review]

**Pull request merged**